add Agent Pull Policy to helm chart

njayp opened the pull request:
## Description

add Agent Pull Policy to helm chart. It is defaulted to "IfNotPresent" in the values file, meaning it will always be present in the traffic manager as an env var

## Checklist

<!--
  Please review the requirements for each checkbox, and check them
  off (change "[ ]" to "[x]") as you verify that they are complete.
-->

 - [x] I made sure to update `./CHANGELOG.yml`.
 - [ ] I made sure to add any docs changes required for my change (including release notes).
 - [ ] My change is adequately tested.
 - [ ] I updated `DEVELOPING.md` with any special dev tricks I had to use to work on this code efficiently.
 - [ ] I updated `TELEMETRY.md` if I added, changed, or removed a metric name.
 - [ ] Once my PR is ready to have integration tests ran, I posted the PR in #telepresence-dev in the datawire-oss slack so that the "ok to test" label can be applied.

njayp committed:
add Agent Pull Policy to helm chart

Signed-off-by: njayp <nickjaypowell@gmail.com>

thallgren commented on charts/telepresence/values.yaml:
```diff
@@ -213,6 +213,7 @@ agent:
     name:
     tag:
     pullSecrets: []
+    pullPolicy: IfNotPresent
```
This should be documented in the chart's README.md.

thallgren commented on cmd/traffic/cmd/manager/managerutil/envconfig.go:
```diff
@@ -52,6 +52,7 @@ type Env struct {
 
 	AgentRegistry             string                      `env:"AGENT_REGISTRY,           parser=nonempty-string"`
 	AgentImage                string                      `env:"AGENT_IMAGE,              parser=string,         default="`
+	AgentImagePullPolicy      string                      `env:"AGENT_IMAGE_PULL_POLICY,  parser=string,		 default="`
```
```suggestion
	AgentImagePullPolicy      string                      `env:"AGENT_IMAGE_PULL_POLICY,  parser=string,         default="`
```
because using tabs inside a string makes the string misaligned unless you have a specific tab-with setting.

njayp committed:
update chart readme

Signed-off-by: njayp <nickjaypowell@gmail.com>

njayp committed:
Merge branch 'release/v2' into njayp/agentPullPolicy

Signed-off-by: njayp <nickjaypowell@gmail.com>

njayp committed:
Merge branch 'release/v2' into njayp/agentPullPolicy

Commit:
GitHub adds HOMEBREW_NO_INSTALL_FROM_API to the env, causing troubles.

This commit removes it by adding the following to the GitHub workflows
that use `brew`:

    env:
        HOMEBREW_NO_INSTALL_FROM_API: ""

and

     brew untap homebrew/core homebrew/cask

Signed-off-by: Thomas Hallgren <thomas@datawire.io>
